From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Spottedleaf <spottedleaf@spottedleaf.dev>
Date: Mon, 18 Jan 2021 10:17:39 -0800
Subject: [PATCH] Don't allow StructureLocateEvent to change worlds

Callers and even the function itself aren't expecting
this to happen

diff --git a/src/main/java/net/minecraft/world/level/chunk/ChunkGenerator.java b/src/main/java/net/minecraft/world/level/chunk/ChunkGenerator.java
index 4a40fa5f7c71481ef0b275955bbd81a737889781..79b02f8c15906d37a185e2a80d0e417364f1d926 100644
--- a/src/main/java/net/minecraft/world/level/chunk/ChunkGenerator.java
+++ b/src/main/java/net/minecraft/world/level/chunk/ChunkGenerator.java
@@ -196,7 +196,7 @@ public abstract class ChunkGenerator {
         // Get origin location (re)defined by event call.
         center = new BlockPos(event.getOrigin().getBlockX(), event.getOrigin().getBlockY(), event.getOrigin().getBlockZ());
         // Get world (re)defined by event call.
-        world = ((org.bukkit.craftbukkit.CraftWorld) event.getOrigin().getWorld()).getHandle();
+        //world = ((org.bukkit.craftbukkit.CraftWorld) event.getOrigin().getWorld()).getHandle(); // Paper - callers and this function don't expect this to change
         // Get radius and whether to find unexplored structures (re)defined by event call.
         radius = event.getRadius();
         skipExistingChunks = event.shouldFindUnexplored();
